contractcourt/channel_arbitrator.go

```diff
@@ -2387,6 +2387,37 @@ func (c *ChannelArbitrator) UpdateContractSignals(newSignals *ContractSignals) {
 	}
 }
 
+// contractUpdateSignal is a struct that carries the latest set of
+// ContractUpdate for a particular key. It also carries a done chan that should
+// be closed by the recipient.
+type contractUpdateSignal struct {
+	// newUpdate contains the latest ContractUpdate for a key.
+	newUpdate *ContractUpdate
+
+	// doneChan is an acknowledgement channel.
+	doneChan chan struct{}
+}
+
+// NotifyContractUpdate notifies the ChannelArbitrator that a new
+// ContractUpdate is available from the link. The link will be paused until
+// this function returns.
+func (c *ChannelArbitrator) NotifyContractUpdate(update *ContractUpdate) {
+	done := make(chan struct{})
+
+	select {
+	case c.htlcUpdates <- &contractUpdateSignal{
+		newUpdate: update,
+		doneChan:  done,
+	}:
+	case <-c.quit:
+	}
+
+	select {
+	case <-done:
+	case <-c.quit:
```

Comment:
Should the quit case return an `ErrShuttingDown` so that we know we didn't actually process the update?

Reply:
If the quit case is taken, then either:
- the channel arbitrator is shutting down due to resolution
- lnd is shutting down

In the first case, the link is already shut down (or about to be, the link also can't do any state updates) and in the second the switch has already shut down along with all links (due to the switch being shutdown before the chain arbitrator). I see your point though, there's not much point in having an error in the function signature if we don't use it

Reply:
I think we need to put a debug log here just to make our life easier for future debugging.

Reply:
Actually I think the error is needed, see comments below.

Reply:
returning a shutdown error, decided to not do the debug log since it will get logged in the link as the link blocks until response